File: Python/queue.py
import json

class Queue():
    def __init__(self):
        self.items = []
    
    def isEmpty(self):
        return self.items == []
    
    def enqueue(self, item):
        self.items.insert(0, item)
    
    def dequeue(self):
        if not self.isEmpty():
            return self.items.pop()
        else:
            raise Exception('Queue is empty!')
    
    def size(self):
        return len(self.items)
    
    def peek(self):
        if not self.isEmpty():
            return self.items[-1]
        else:
            raise Exception('Queue is empty!')
    
    def toJson(self):
        return json.dumps(list(reversed(self.items)))

def main():
    queue = Queue()
    queue.enqueue('dog')
    queue.enqueue('cat')
    queue.enqueue('mouse')
    print(queue.peek())
    print(queue.size())
    print(queue.toJson())
    print(queue.dequeue())
    print(queue.toJson())
    print(queue.dequeue())
    print(queue.peek())
    print(queue.toJson())

if __name__ == "__main__":
    main()      

File: Javascript/queue.js
function Queue() {
    let items = [];
    this.enqueue = function (element) {
        items.push(element);
    }
    this.dequeue = function () {
        return items.shift();
    }
    this.front = function () {
        return items[0];
    }
    this.isEmpty = function () {
        return items.length === 0;
    }
    this.size = function () {
        return items.length;
    }
    this.print = function () {
        console.log(items.toString());
    }
}

let queue = new Queue()
queue.enqueue(2)
queue.enqueue(3)
queue.enqueue(1)
queue.enqueue(6)
queue.enqueue(8)
queue.print()
console.log(queue.front())
console.log(queue.isEmpty())
console.log(queue.size())
queue.dequeue()
queue.print()
console.log(queue.front())
console.log(queue.isEmpty())
console.log(queue.size())
queue.dequeue()
queue.print()
